### PR TITLE
Added Azure.MariaDB.AllowAzureAccess

### LINKED ISSUE #1857: Azure Database for MariaDB servers should disable allow Azure services on firewall
# Rule request

## Suggested rule change

Azure Database for MariaDB servers should disable allow Azure services on firewall.

Allow access to Azure services, permits any Azure service including other Azure customers, network based-access to databases on the same MariaDB server instance. 

It's not really optimal, even tho you still need the valid JWT token.

Security pillar for this one.

## Applies to the following

The rule applies to the following:

- Resource type: **[Microsoft.DBforMariaDB/servers]**

## Additional context

[Connecting from Azure](https://learn.microsoft.com/azure/mariadb/howto-manage-firewall-portal#connecting-from-azure)
[Template reference](https://learn.microsoft.com/azure/templates/microsoft.dbformariadb/servers)

### PR DESCRIPTION
**Pull request by BenjaminEngeset**
## PR Summary

Fixes #1857 

Added Azure.MariaDB.AllowAzureAccess rule.

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [x] Unit tests created/ updated
  - [x] Rule documentation created/ updated
  - [x] Link to a filed issue
  - [x] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

**Comment by BenjaminEngeset** [comment]
@BernieWhite 

To be reviewed when you have the time.

**Review by BernieWhite — CHANGES_REQUESTED** [review]
Let's double check `GetMariaDBFirewallRule` and add some additional test cases because I don't think it is working as intended.

Otherwise nice work @BenjaminEngeset.

File: src/PSRule.Rules.Azure/rules/Azure.MariaDB.Rule.ps1 [hunk 1]
```diff
@@ -168,4 +176,19 @@ function global:GetMariaDBVNETRuleName {
...
 }
 
+function global:GetMariaDBFirewallRule {
+    [CmdletBinding()]
+    [OutputType([PSObject])]
+    param ()
+    process {
+        if ($PSRule.TargetType -eq 'Microsoft.DBforMariaDB/servers') {
+            GetSubResources -ResourceType 'Microsoft.DBforMariaDB/servers/firewallRules' |
+            ForEach-Object { $_ }
```
[the remark this answers is not in the document]
Comment (BenjaminEngeset) [reply src/PSRule.Rules.Azure/rules/Azure.MariaDB.Rule.ps1]: You are correct.

File: src/PSRule.Rules.Azure/rules/Azure.MariaDB.Rule.ps1 [hunk 2]
```diff
@@ -168,4 +176,19 @@ function global:GetMariaDBVNETRuleName {
...
+        }
+        elseif ($PSRule.TargetType -eq 'Microsoft.DBforMariaDB/servers/firewallRules') {
+            $PSRule.TargetName
```
[the remark this answers is not in the document]
Comment (BenjaminEngeset) [reply src/PSRule.Rules.Azure/rules/Azure.MariaDB.Rule.ps1]: Yes. Mistake from my side.

**Comment by BenjaminEngeset** [comment]
@BernieWhite 

Check the latest updates. Did I understand you correctly or do you want me to create tests where we declare rules outside the parent resource?

**Comment by BenjaminEngeset** [comment]
@BernieWhite 

Updated. Could you check it now and see if we're alligned?

**Review by BernieWhite — APPROVED** [review]
Great work @BenjaminEngeset. The updates look good. All good to merge.

**Pull request merged**